mentat/code_file_manager.py

```diff
@@ -72,23 +72,34 @@ def get_code_message(self):
 
         return "\n".join(code_message)
 
+    def _add_file(self, abs_path):
+        logging.info(f"Adding new file {abs_path} to context")
+        self.code_context.files[abs_path] = CodeFile(abs_path)
+        # create any missing directories in the path
+        abs_path.parent.mkdir(parents=True, exist_ok=True)
+
+    def _delete_file(self, abs_path: Path):
+        logging.info(f"Deleting file {abs_path}")
+        if abs_path in self.code_context.files:
+            del self.code_context.files[abs_path]
+        abs_path.unlink()
+
     def _handle_delete(self, delete_change):
-        file_path = self.config.git_root / delete_change.file
-        if not file_path.exists():
-            logging.error(f"Path {file_path} non-existent on delete")
+        abs_path = self.config.git_root / delete_change.file
+        if not abs_path.exists():
+            logging.error(f"Path {abs_path} non-existent on delete")
             return
```

Comment:
This is actually necessary; abspath appends the CWD to the beginning of the path, which might not necessarily be the current git repo (it could be a subdirectory). This does remind me though, I don't think we have any tests where we run mentat from a subdiretory of the git repo it's actually in; we should probably think about adding some of those!

Reply:
> I don't think we have any tests where we run mentat from a subdiretory of the git repo it's actually in; we should probably think about adding some of those!

@PCSwingle can you add an issue for this